### PR TITLE
Add CI for python lightning app Python unit tests

### PR DESCRIPTION
**Pull request by manskx**
## What does this PR do?

Fixes #\<issue_number>

### Does your PR introduce any breaking changes? If yes, please list them.

Adding CI for lightning app tests and update tests
## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda

### CODE REVIEW

File: .github/workflows/ci-app_tests.yml [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+on: # Trigger the workflow on push or pull request, but only for the master branch
+  push:
+    branches:
+      - "master"
+  pull_request:
+    paths:
+      - "src/lightning_app/**"
+      - "tests/tests_app/**"
+      - "requirements/app/**"
+      - "setup.py"
```
Comment (Borda) [inline .github/workflows/ci-app_tests.yml:13]: with this we can't set it as required check

File: .github/workflows/ci-app_tests.yml [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Set min. dependencies
+        if: matrix.requires == 'oldest'
+        run: |
```
Comment (Borda) [inline .github/workflows/ci-app_tests.yml:41]: lets use https://github.com/Lightning-AI/lightning/blob/f65aa7471921ab78f2ed131d3692507b1a06e3de/.actions/assistant.py#L66
check existing test full CI workflow

File: .github/workflows/ci-app_tests.yml [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+      - name: Install Lightning as top-level
+        run: pip install -e . -r requirements/app/base.txt
+        shell: bash
+
+      - name: Tests
+        working-directory: ./tests
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
```
Comment (Borda) [inline .github/workflows/ci-app_tests.yml:95]: need to be added, @lexierule

File: .github/workflows/ci-app_tests.yml (hunk 2 above)
Comment (manskx) [reply .github/workflows/ci-app_tests.yml:41]: can we do it in the following PR ?

File: .github/workflows/ci-app_tests.yml (hunk 1 above)
Comment (manskx) [reply .github/workflows/ci-app_tests.yml:13]: How should we do it ?

File: .github/workflows/ci-app_tests.yml [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+          restore-keys: ${{ runner.os }}-pip-${{ matrix.python-version }}-${{ matrix.requires }}-td${{ steps.times.outputs.period }}-
+
+      - name: Install dependencies
+        run: |
+          pip --version
+          pip install -r requirements/app/devel.txt --quiet --find-links https://download.pytorch.org/whl/cpu/torch_stable.html
+          pip list
+        shell: bash
+
+#      - name: Start Redis
```
Comment (tchaton) [inline .github/workflows/ci-app_tests.yml:73]: Why did we remove this ?

**Review by tchaton — APPROVED** [review]
Awesome work !

File: .github/workflows/ci-app_tests.yml (hunk 4 above)
Comment (manskx) [reply .github/workflows/ci-app_tests.yml:73]: wasn't it already removed?

Comment (Borda) [reply .github/workflows/ci-app_tests.yml:73]: lets drop the comment completely

**Review by awaelchli — APPROVED** [review]

**Pull request merged**